Penteract/Assets/Scripts/LaserTurret.cpp

```diff
@@ -0,0 +1,97 @@
+#include "LaserTurret.h"
+
+#include "GameplaySystems.h"
+#include "GameObject.h"
+
+EXPOSE_MEMBERS(LaserTurret) {
+    MEMBER(MemberType::FLOAT, coolDownOn),
+    MEMBER(MemberType::FLOAT, coolDownOnTimer),
+    MEMBER(MemberType::FLOAT, coolDownOff),
+    MEMBER(MemberType::FLOAT, coolDownOffTimer),
+    MEMBER(MemberType::GAME_OBJECT_UID, laserTargetUID)
+};
+
+GENERATE_BODY_IMPL(LaserTurret);
+
+void LaserTurret::Start() {
+	
+    GameObject* owner = &GetOwner();
+
+    if (owner) {
+        animationComp = owner->GetComponent<ComponentAnimation>();
+    }
+
+    laserObject = GameplaySystems::GetGameObject(laserTargetUID);
+    if (laserObject) {
+        laserObject->Disable();
+    }
+
+}
+
+void LaserTurret::Update() {
+
+    if (!animationComp) {
+        Debug::Log("animationComp is nullptr");
+    }
+
+    switch (currentState)
+    {
+    case TurretState::IDLE_START:
+        if (coolDownOffTimer <= coolDownOff) {
+            coolDownOffTimer += Time::GetDeltaTime();
+            if (coolDownOffTimer > coolDownOff) {
+                coolDownOffTimer = 0;
+                currentState = TurretState::START;
+                animationComp->SendTrigger(states[static_cast<unsigned int>(TurretState::IDLE_START)] + states[static_cast<unsigned int>(TurretState::START)]);
```

Comment:
You could use currentState as an int and convert the enum class into an enum so you don't need to cast everytime, but i'm just being picky here

Reply:
https://youtu.be/1taLXjVxB4U?t=71

> 
> 
> You could use currentState as an int and convert the enum class into an enum so you don't need to cast everytime, but i'm just being picky here